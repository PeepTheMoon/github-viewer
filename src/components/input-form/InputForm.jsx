import React from 'react';
import PropTypes from 'prop-types';

const InputForm = ({ username, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={username} onChange={onChange} />
    <button>Submit</button>
  </form>
);

InputForm.propTypes = {
  username: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default InputForm;
