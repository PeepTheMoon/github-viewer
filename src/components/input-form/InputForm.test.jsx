import React from 'react';
import { shallow } from 'enzyme';
import InputForm from './InputForm';

describe('InputForm component', () => {
  it('renders App', () => {
    const wrapper = shallow(<InputForm username="PeepTheMoon" onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
