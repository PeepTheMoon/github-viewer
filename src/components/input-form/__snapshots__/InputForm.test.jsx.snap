// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm component renders App 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="PeepTheMoon"
  />
  <button>
    Submit
  </button>
</form>
`;
